### PR TITLE
Secondmillestone

### PR DESCRIPTION
**Pull request by josuerushanika**
In this PR i did:

- [ ] Created Nameable class.
- [ ] Implemented correct_name method that raises a NotImplementedError.
- [ ] Made the Person class inherits from Nameable.
- [ ] Implemented correct_name method.
- [ ] Created base Decorator class inherits from Nameable.
- [ ] In the constructor, assigned a nameable object from params to an instance variable.
- [ ] Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
- [ ] Created CapitalizeDecorator and TrimmerDecorator classes:

                      **For the CapitalizeDecorator:**

- [ ] Created a class that inherits from the base Decorator class.
- [ ] Implemented a method correct_name that capitalizes the output of @nameable.correct_name.

**For the TrimmerDecorator:**
- [ ] Created a class that inherits from the base Decorator class.
- [ ] Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 
characters. If it's longer it should trim the word.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -13,6 +17,10 @@ def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age?
```
Comment (BenMukebo) [inline person.rb:26]: - [ ] Greate job so far 👏🏻 However according to the project requirements, you should implement decorator and run a program using the command line. Actually, I'm getting this error shown above 👇🏻 Kidly solve this 👍🏻 

image

- You may use this code in the `person` class to check if you managed to decorate a class:
```ruby
  person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = Capitalize.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = ...
   ...
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @josuerushanika,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Good commit messages ✔️ 
- Professional README ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (josuerushanika) [reply person.rb:26]: @BenMukebo  Thanks it's fixed

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @josuerushanika👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Gitflow is followed. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**